feature/interface/aggregate/otg_tests/aggregate_forwarding_viable_test/aggregate_forwarding_viable_test.go

```diff
@@ -0,0 +1,596 @@
+		srci.GetOrCreateEthernet().PortSpeed = fptest.GetIfSpeed(t, srcp)
+	}
+	fptest.LogQuery(t, srcp.String(), srciPath.Config(), srci)
+	gnmi.Replace(t, tc.dut, srciPath.Config(), srci)
+	if *deviations.ExplicitInterfaceInDefaultVRF {
+		fptest.AssignToNetworkInstance(t, tc.dut, srcp.Name(), *deviations.DefaultNetworkInstance, 0)
+	}
+
+	for _, port := range tc.dutPorts[1:] {
+		i := &oc.Interface{Name: ygot.String(port.Name())}
+		i.Type = ethernetCsmacd
+
+		if *deviations.InterfaceEnabled {
+			i.Enabled = ygot.Bool(true)
+		}
+
+		tc.configDstMemberDUT(i, port)
+		iPath := d.Interface(port.Name())
+		if *deviations.ExplicitPortSpeed {
+			i.GetOrCreateEthernet().PortSpeed = fptest.GetIfSpeed(t, port)
+		}
+		fptest.LogQuery(t, port.String(), iPath.Config(), i)
+		gnmi.Replace(t, tc.dut, iPath.Config(), i)
+	}
+}
+
+// configureATE configures source and destination port of ATE and add creates port-channel as well.
+func (tc *testArgs) configureATE(t *testing.T) {
+	t.Helper()
+	if len(tc.atePorts) < 2 {
+		t.Fatalf("Testbed requires at least 2 ports, got: %v", tc.atePorts)
+	}
+
+	p0 := tc.atePorts[0]
+	tc.top.Ports().Add().SetName(p0.ID())
+	d0 := tc.top.Devices().Add().SetName(ateSrc.Name)
+	srcEth := d0.Ethernets().Add().SetName(ateSrc.Name + ".Eth").SetMac(ateSrc.MAC)
+	srcEth.Connection().SetChoice(gosnappi.EthernetConnectionChoice.PORT_NAME).SetPortName(p0.ID())
+	srcEth.Ipv4Addresses().Add().SetName(ateSrc.Name + ".IPv4").SetAddress(ateSrc.IPv4).SetGateway(dutSrc.IPv4).SetPrefix(int32(ateSrc.IPv4Len))
+	srcEth.Ipv6Addresses().Add().SetName(ateSrc.Name + ".IPv6").SetAddress(ateSrc.IPv6).SetGateway(dutSrc.IPv6).SetPrefix(int32(ateSrc.IPv6Len))
+
+	// Adding the rest of the ports to the configuration and to the LAG
+	agg := tc.top.Lags().Add().SetName(ateDst.Name)
+	if tc.lagType == lagTypeSTATIC {
+		lagId, _ := strconv.Atoi(tc.aggID)
+		agg.Protocol().SetChoice("static").Static().SetLagId(int32(lagId))
+		for i, p := range tc.atePorts[1:] {
+			port := tc.top.Ports().Add().SetName(p.ID())
+			newMac, err := incrementMAC(ateDst.MAC, i+1)
+			if err != nil {
+				t.Fatal(err)
+			}
+			agg.Ports().Add().SetPortName(port.Name()).Ethernet().SetMac(newMac).SetName("LAGRx-" + strconv.Itoa(i))
+		}
+	} else {
+		agg.Protocol().SetChoice("lacp")
+		for i, p := range tc.atePorts[1:] {
+			port := tc.top.Ports().Add().SetName(p.ID())
+			newMac, err := incrementMAC(ateDst.MAC, i+1)
+			if err != nil {
+				t.Fatal(err)
+			}
+			lagPort := agg.Ports().Add().SetPortName(port.Name())
+			lagPort.Ethernet().SetMac(newMac).SetName("LAGRx-" + strconv.Itoa(i))
+			lagPort.Lacp().SetActorActivity("active").SetActorPortNumber(int32(i) + 1).SetActorPortPriority(1).SetLacpduTimeout(0)
+		}
+	}
+
+	dstDev := tc.top.Devices().Add().SetName(agg.Name())
+	dstEth := dstDev.Ethernets().Add().SetName(ateDst.Name + ".Eth").SetMac(ateDst.MAC)
+	dstEth.Connection().SetChoice(gosnappi.EthernetConnectionChoice.LAG_NAME).SetLagName(agg.Name())
```

Comment:
Same as above, don't believe SetChoice is necessary.

Reply:
Ethernet.Connection Set choice is mandatory

Reply:
Latest gosnappi (https://github.com/open-traffic-generator/snappi/blob/main/gosnappi/gosnappi.go at https://github.com/open-traffic-generator/snappi/commit/05c4a89a906eb0d9e842f0d9a5dba56114a3e81d) does set the choice on the 'Connection' object automatically (line 60736 for the function).

Reply:
Removed SetChoice